src/components/NewEventDialog.js

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import useForm from 'react-hook-form';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { TextField, Button, Grid, Box } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import gql from 'graphql-tag';
+
+import useStyle from './UseStyle';
+
+const mutation = gql`
+  mutation addEvent(
+    $title: String!
+    $start: String!
+    $end: String!
+    $description: String!
+  ) {
+    addEvent(
+      title: $title
+      start: $start
+      end: $end
+      description: $description
+    ) {
+      id
+      title
+      start
+      end
+      description
+    }
+  }
+`;
+
+function NewEventDialog(props) {
+  const { onClose, open, selectedDate } = props;
+  const classes = useStyle();
+
+  const { handleSubmit, register, errors } = useForm();
+  const [addEvent, { loading, error }] = useMutation(mutation);
+
+  const onSubmit = useCallback(
+    ({ title, description, date, startTime, endTime }) => {
```

Comment:
Just noticed there's no validation that `endTime` is after `startTime` which is confusing.

Reply:
Done